version: "3.8"

services:
  db:
    image: mysql:8
    container_name: mysql-container
    volumes:
      - db-data:/var/lib/mysql
      - ./dump/plutocinema.sql:/docker-entrypoint-initdb.d/01.sql:ro
    ports:
      - "3306:3306"
    networks:
      - app-network
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: plutocinema
      MYSQL_ROOT_HOST: "%"

  python-seeder:
    image: python:3.9-slim
    build:
      context: .
      dockerfile: Dockerfile-python
    volumes:
      - ./database:/app/database
      - ./database/result_cleanse:/app/result_cleanse
    environment:
      DB_HOST: db
      DB_USER: root
      DB_PASSWORD: root
      DB_NAME: plutocinema
    networks:
      - app-network
    depends_on:
      - db
    restart: "no"
    command: python /app/database/seeder.py > /dev/null 2>&1
    deploy:
      replicas: 0

  client:
    build:
      context: .
      dockerfile: Dockerfile-client
    ports:
      - "4173:4173"
    networks:
      - app-network
    env_file:
      - view/.env

  server:
    build:
      context: .
      dockerfile: Dockerfile-server
    ports:
      - "5000:5000"
    networks:
      - app-network
    depends_on:
      - db
    environment:
      DB_HOST: db
      DB_USER: root
      DB_PASS: root
      DB_NAME: plutocinema
      SERVER_PORT: 5000
      SERVER_URL: http://localhost:5000
      CLIENT_URL: http://localhost:4173
    env_file:
      - server/.env
networks:
  app-network:
    driver: bridge

volumes:
  db-data:
